third_party/move/move-compiler-v2/tests/checking/dotdot/dotdot_valid.move

```diff
@@ -0,0 +1,104 @@
+module 0x42::test {
+    struct S0() has copy;
+
+    struct S1(u8) has copy;
+
+    struct S2(bool, S0) has copy;
+
+    struct S3 has copy {
+        x: bool,
+        y: u8
+    }
+
+    struct S4<T> has copy {
+        x: T,
+        y: S3
+    }
+
+    enum E1 has drop {
+        A(u8, bool),
+        B(u8),
+    }
+
+    fun simple_0(x: S0) {
+        let S0(..) = x;
```

Comment:
These bind matches also occur in assigments:
```
    S0(..) = x;
```
and lambdas:
```
    inline fun read_s0(x: S0, y:|S0|u64) {
        y(x)
    }
    ...
        read_s0(x, |S0(..)| 3);
```
You should add such cases for all of your different match cases, ideally.

Actually, more plausible cases are:
```
    S2(y, ..) = x;
```
and
```
    read_s2(x, |S2(y, ..)|y);
```

Reply:
1. Both positional fields and ".."s are not supported yet in the LHS of an assignment. Supporting it would require a lot of changes. This is because assignments are handled differently than declarations and matches in the parser AST.
```
Assign(Box<Exp>, Box<Exp>),
```
where the LHS is an `Exp` instead of a `BindList`. We have to make sure ".." only appear in an `Exp` if the expression is in the LHS. Ensuring this either statically or dynamically requires some effort. Given the time budget, and how rare this feature is used, I think we should defer this. Submitted a feature request https://github.com/aptos-labs/aptos-core/issues/14295

2. Tested in `third_party/move/move-compiler-v2/tests/checking/dotdot/dotdot_valid.move:test_lambda_param`

Reply:
Can you add a test to show what error is generated for the assignment case, then?

Reply:
Done. third_party/move/move-compiler-v2/tests/checking/dotdot/assign.move

Reply:
done